wagtail/wagtailcore/models.py

```diff
@@ -537,15 +537,23 @@ def check(cls, **kwargs):
         from wagtail.wagtailadmin.forms import WagtailAdminPageForm
         if not issubclass(cls.base_form_class, WagtailAdminPageForm):
             errors.append(checks.Error(
-                "base_form_class does not extend WagtailAdminPageForm",
+                "{}.base_form_class does not extend WagtailAdminPageForm".format(
+                    cls.__name__),
                 hint="Ensure that {}.{} extends WagtailAdminPageForm".format(
                     cls.base_form_class.__module__,
                     cls.base_form_class.__name__),
                 obj=cls,
                 id='wagtailcore.E002'))
-        # Sadly, there is no way of checking the form class returned from
-        # cls.get_edit_handler().get_form_class(cls), as these calls can hit
-        # the DB in order to fetch content types.
+
+        edit_handler = cls.get_edit_handler()
+        if not issubclass(edit_handler.get_form_class(cls), WagtailAdminPageForm):
```

Comment:
I am pretty sure this check is now fine - all the tests still pass, with no modifications, which is a good sign. Previously the check was not possible because some form fields/edit handlers such as PageChooserPanel and SnippetChooserPanel used content types to limit choices. #1950 and #2015 (and possibly others) made it so that content types are no longer used. Plain model classes are used instead. This means that constructing a form class should no longer hit the database.